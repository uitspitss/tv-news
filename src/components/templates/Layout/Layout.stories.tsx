import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import base from 'paths.macro';
import tw from 'twin.macro';

import { Layout, LayoutProps } from './Layout';

const path = base.replace(
  /^.+\/(atoms|molecules|organisms|templates|pages)\/(.+)\//,
  '$1/$2',
);

export default {
  title: `${path}`,
  component: Layout,
} as Meta;

const Template: Story<LayoutProps> = (args) => (
  <Layout {...args}>
    <div tw="bg-gray-500 p-4">MAIN CONTENT</div>
  </Layout>
);

export const Desktop = Template.bind({});

export const Mobile = Template.bind({});
Mobile.parameters = {
  viewport: {
    defaultViewport: 'iphonex',
  },
};
