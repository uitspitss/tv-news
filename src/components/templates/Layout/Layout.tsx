import React, { memo, FC, ReactNode } from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import tw, { styled } from 'twin.macro';

import { Header } from '~/components/organisms/Header';
import { Footer } from '~/components/organisms/Footer';

export type LayoutProps = {
  /**
   * class for twin.macro
   */
  className?: string;
  /**
   * children
   */
  children: ReactNode;
};

type Props = LayoutProps;

const Component: FC<Props> = (props) => {
  const { children, className } = props;

  return (
    <div className={className}>
      <header>
        <Header className="header" />
      </header>
      <main>{children}</main>
      <footer>
        <Footer />
      </footer>
    </div>
  );
};

const StyledComponent = styled(Component)`
  & > header {
    ${tw`sticky top-0 left-0 w-full h-11`}
  }

  & > main {
    height: calc(100% - 44px - 36px);
    min-height: calc(100vh - 44px - 36px);
  }
`;

/**
 * レイアウト
 */
export const Layout: FC<LayoutProps> = memo((props) => (
  <StyledComponent {...props} />
));
