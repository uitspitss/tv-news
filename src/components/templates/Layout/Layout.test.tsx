import React from 'react';
import { render } from '@testing-library/react';

import { Layout } from './Layout';

describe('Layout', () => {
  test('snapshot test', () => {
    const { container } = render(
      <Layout>
        <a href="/">link</a>
      </Layout>,
    );
    expect(container).toMatchSnapshot();
  });
});
