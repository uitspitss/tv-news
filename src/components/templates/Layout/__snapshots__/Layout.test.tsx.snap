// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot test 1`] = `
.emotion-0>header {
  -webkit-position: sticky;
  position: sticky;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 2.75rem;
}

.emotion-0>main {
  height: calc(100% - 44px - 36px);
  min-height: calc(100vh - 44px - 36px);
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.emotion-2 {
  --tw-bg-opacity: 1;
  background-color: rgba(165, 180, 252, var(--tw-bg-opacity));
  margin-bottom: 0.5rem;
}

@media (min-width: 1024px) {
  .emotion-2 {
    margin-bottom: 1rem;
  }
}

.emotion-2>nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width: 1024px) {
  .emotion-2>nav {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-2>nav>.title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(243, 244, 246, var(--tw-text-opacity));
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.75rem;
  letter-spacing: -0.025em;
}

.emotion-2>nav>.toggleButton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width: 1024px) {
  .emotion-2>nav>.toggleButton {
    display: none;
  }
}

.emotion-2>nav>.menuList {
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: none;
}

@media (min-width: 1024px) {
  .emotion-2>nav>.menuList {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: auto;
  }
}

@media (min-width: 1024px) {
  .emotion-2>nav>.menuList {
    display: block;
  }
}

.emotion-4 {
  font-size: 1rem;
  line-height: 1.5rem;
  border-radius: 0.25rem;
  border-style: none;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.emotion-6 {
  font-size: 0.875rem;
  line-height: 1.25rem;
}

@media (min-width: 1024px) {
  .emotion-6 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-6>a {
  display: block;
  margin-top: 1rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

@media (min-width: 1024px) {
  .emotion-6>a {
    display: inline-block;
    margin-top: 0px;
  }
}

.emotion-6>a:not(:last-child) {
  margin-right: 1rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width: 640px) {
  .emotion-8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width: 1024px) {
  .emotion-8 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-8>p {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

.emotion-8>p>a {
  --tw-text-opacity: 1;
  color: rgba(75, 85, 99, var(--tw-text-opacity));
  margin-left: 0.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <header>
      <div
        class="header emotion-2 emotion-3"
      >
        <nav>
          <div
            class="title"
          >
            <a
              href="/"
            >
              *WEBSITE TITLE*
            </a>
          </div>
          <div
            class="toggleButton"
          >
            <button
              class="emotion-4 emotion-5"
              type="button"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="closed"
                  d="M4 6h16M4 12h16M4 18h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
          <div
            class="menuList"
          >
            <div
              class="emotion-6 emotion-7"
            >
              <a
                href="/users-ssr"
              >
                Users SSR
              </a>
              <a
                href="/users-ssg"
              >
                Users SSG
              </a>
              <a
                href="/users-swr"
              >
                Users useSWR
              </a>
              <a
                href="/users-recoil"
              >
                Users useUsers
              </a>
            </div>
          </div>
        </nav>
      </div>
    </header>
    <main>
      <a
        href="/"
      >
        link
      </a>
    </main>
    <footer>
      <div
        class="emotion-8 emotion-9"
      >
        <p>
          ©
          2020 - 2021
           *WEBSITE TITLE* —
          <a
            href="https://twitter.com/uitspitss"
            rel="noopener noreferrer"
            target="_blank"
          >
            @uitspitss
          </a>
        </p>
      </div>
    </footer>
  </div>
</div>
`;
